server/handlers/get_totp.go

```diff
@@ -0,0 +1,35 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/keratin/authn-server/app"
+	"github.com/keratin/authn-server/app/services"
+	"github.com/keratin/authn-server/lib/route"
+	"github.com/keratin/authn-server/server/sessions"
+)
+
+//GetTOTP begins the TOTP onboarding process
+func GetTOTP(app *app.App) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// check for valid session with live token
+		accountID := sessions.GetAccountID(r)
+		if accountID == 0 {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		img, err := services.TOTPCreator(app.AccountStore, app.TOTPCache, accountID, route.MatchedDomain(r))
```

Comment:
This has been a question mark for me [since the original PR](https://github.com/keratin/authn-server/pull/133#discussion_r331208949), because it means that a GET request is generating server state. It also involves create a new TOTPCache, adding `.write()` semantics to the BlobStore, etc., and I wonder if it could be simpler.

What if GetTOTP generated a new secret every time, and returned it in a JWT alongside the QR code? The PostTOTP endpoint would validate the JWT, extract its secret, and use that to verify the user has a working code generator.

Thoughts? It feels consistent with how AuthN handles other processes. In fact, it's similar enough to password resets that I wonder if we would need to imitate the locking mechanism there and ensure that no matter how many JWTs are generated, using one will expire the rest.